lib/realtime/server.ex

```diff
@@ -9,13 +9,10 @@ defmodule Realtime.Server do
   use GenServer
 
   alias Gtfs.Route
+  alias Realtime.Vehicles
   require Logger
 
-  @type vehicles_by_route_id :: %{optional(Route.id()) => [Realtime.Vehicle.t()]}
-
-  @type state :: %{
-          vehicles: vehicles_by_route_id()
-        }
+  @typep state :: Route.by_id(Vehicles.for_route())
```

Comment:
Do the server tests need to be updated to reflect all these changes?

Reply:
Ooh, thanks, yeah. I don't know how they were even passing.